src/transporters.ts

```diff
@@ -95,6 +98,13 @@ export class DefaultTransporter {
       }
     }
 
+    // If the user configured an `HTTPS_PROXY` environment variable, create
+    // a custom agent to proxy the request.
+    const proxy = process.env.HTTPS_PROXY || process.env.https_proxy;
+    if (proxy) {
+      opts.httpsAgent = new HttpsProxyAgent(proxy);
```

Comment:
If/Once client libraries start using axios, they will need the same pattern. Should we make an axios wrapper that everyone can use instead of axios?